### PR TITLE
Custom pixelpact hook

### PR DESCRIPTION
**Pull request by mloydd**

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -9,10 +9,12 @@
     "format": "npx prettier --write ."
   },
   "devDependencies": {
+    "@playwright/test": "^1.35.1",
     "pino-pretty": "^10.0.0",
     "prettier": "2.8.8"
   },
   "dependencies": {
+    "@pixelpact-hook/playwright-js": "file:../pixelpact-hooks/playwright-js",
```
Comment (raphiz) [inline example/package.json:17]: Let's discuss the naming of this again.
I'm not sure about how intuitive the word `hook` is. It's basically a client library - one per eco system (js, java, go, ...)
With that in mind, a simpler package name would be sufficient as there won't be any other npm packages, right?

File: example/tests/example.spec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  viewport: { width: 1920, height: 1024 },
+  locale: "de-CH",
+  video: "on",
+  trace: "on",
+  reporter: "html",
+});
+
+import fs from "fs/promises";
+
+test("test login page", async ({ page, browser }, testInfo) => {
```
Comment (raphiz) [inline example/tests/example.spec.js:14]:
```suggestion
test("verify site visually", async ({ page, browser }, testInfo) => {
```

File: pixelpact-hooks/playwright-js/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "pixelpact"
+  },
+  "author": "",
+  "license": "ISC"
```
Comment (raphiz) [inline pixelpact-hooks/playwright-js/package.json:15]: license is `MIT`

File: pixelpact-hooks/playwright-js/package.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "@pixelpact-hook/playwright-js",
+  "version": "1.0.0",
+  "description": "Pixelpact Playwright-JS helps to easily hook into Pixelpact Service",
+  "type": "module",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
```
Comment (raphiz) [inline pixelpact-hooks/playwright-js/package.json:10]: Is there a benefit to this `repository` section?

File: example/package.json (hunk 1 above)
Comment (raphiz) [reply example/package.json:17]: For example percy.io calls it: [`Playwright client library for visual testing with Percy`](https://github.com/percy/percy-playwright)

File: pixelpact-hooks/playwright-js/index.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    flag: "r",
+  })
+);
+const folderPath = getFolderPath();
+
+const expectedFileSuffix = "expected";
+const actualFileSuffix = "actual";
+const diffFileSuffix = "diff";
+
+export async function toMatchVisually(page, testInfo, fileNamePrefix) {
```
Comment (raphiz) [inline pixelpact-hooks/playwright-js/index.js:18]: I think we can get rid of the `testInfo` (maybe in another PR) as follows:

Register a [`test.beforeEach`](https://playwright.dev/docs/api/class-test#test-before-each) hook, that stores the reference to `testInfo` somewhere (eg. a global in the module or maybe even AsyncLocalStorage). The  `toMatchVisually` can then obtain the `testInfo` instance from there, hiding this implementation detail from our public API.

We could also derive the `fileNamePrefix` from the `testInfo.title` if it is not specified.

File: pixelpact-hooks/playwright-js/package.json (hunk 4 above)
Comment (mloydd) [reply pixelpact-hooks/playwright-js/package.json]: I needed to integrate with example test. It's not needed anymore.

**Pull request merged**